### PR TITLE
Matrikelnummer: 0848451

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Matrikelnummer: 0848451

File: ex1/dates_times.py [hunk 1]
```diff
@@ -29,16 +29,41 @@
...
+
+
 # Define a function named how_long that takes two datetime objects dt and ref
 # where ref is the reference datetime, calculates the difference between them and
 # returns the difference as a string formatted like:
 # "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
 # If ref is before dt then use 'since' instead of 'until'
+
+def how_long(dt, ref):
+	if (dt>=ref) :
```
Comment (cpaulik) [inline ex1/dates_times.py:59]: There is no need for brackets here.

File: ex1/exceptions.py [hunk 2]
```diff
@@ -37,6 +37,11 @@
 # Use the function search_n from the functions module inside a new function
 # also named search_n. The function should do the same as functions.search_n
 # but if the variable is not found in the list then raise a ValueError.
+def search_n(ls, x):
+	if functions.search_n(ls, x) == (None, None):
```
Comment (cpaulik) [inline ex1/exceptions.py:41]: Using `==` when comparing with Python Singletons is not recommended.
If you are interested why that is then please see the answers [here](http://stackoverflow.com/questions/2209755/python-operation-vs-is-not)

File: ex1/functions.py [hunk 3]
```diff
@@ -90,26 +118,49 @@ def f():
...
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
+def search_list(a, b):
+	d = {}
+	for i in b:
+		if(i in a):
+			d.update({a.index(i): i})
```
Comment (cpaulik) [inline ex1/functions.py:145]: It is fine like it is but you could also just assign a value to a new key. I find that more readable.

``` python
d[a.index(i)] = i
```

File: ex1/functions.py [hunk 4]
```diff
@@ -90,26 +118,49 @@ def f():
...
+	return d
 
 # Define a function named dict_to_string that takes a dictionary and a
 # separator string. The function should only take elements out of the
 # dictionary whose value is a string and then return a single string containing
 # the strings stored in the dictionary seperated by the separator string.
 # Return an empty string if there are no strings in the dictionary.
+def dict_to_string(di, sep):
+	strLs = [k for k in di.values() if type(k) == type('')]
```
Comment (cpaulik) [inline ex1/functions.py:154]: For type comparison you can use the builtin types like `str` or `int`.

**Comment by cpaulik** [comment]
The code looks fine. But there are no docstrings for any of them. Please compare to https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20 if any comments I made are unclear.

**Pull request closed without merging**